### PR TITLE
User info

### PR DESCRIPTION
**Pull request by EduBeghe**
- Removed functionality that gets userInfo for every request made to the backend
- Added functionality that checks JWT ID Token for user info on every call made to the backend
- Added functionality that checks JWT ID Token signature, so that every token we get is valid
- Added certificate.pem file to the backend, necessary to decode ID Token
- Conditionally added UserInfo token to each request from the frontend ( Why conditionally? Because Auth0 rejects every request with custom headers as UserInfo )

### CODE REVIEW

File: backend/certificate.pem [hunk 1]
```diff
@@ -0,0 +1,19 @@
+-----BEGIN CERTIFICATE-----
+MIIDEzCCAfugAwIBAgIJVOQEFdoBdN8MMA0GCSqGSIb3DQEBCwUAMCcxJTAjBgNV
```
Comment (MichaelAshmead) [inline backend/certificate.pem:2]: Definitely don't want to commit this, can you remove from PR?

File: frontend/src/app/services/token.interceptor.ts [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
+    let host = request.url.substr(8,28);
+    
+    if (host != environment.auth0_domain) {
+      let accessToken = localStorage.getItem("access_token");
+      let idToken = localStorage.getItem("id_token");
       request = request.clone({
         setHeaders: {
-          // Injecting the AuthService caused a circular dependency, so for now just getting the id_token "manually"
-          Authorization: `Bearer ${localStorage.getItem("id_token")}`
+          Authorization: `Bearer ${accessToken}`,
```
Comment (MichaelAshmead) [inline frontend/src/app/services/token.interceptor.ts:19]: Is there any need to send the accessToken to the backend anymore?

File: backend/lib/middleware/userInfo.ts [hunk 3]
```diff
@@ -1,25 +1,31 @@
...
+        req.body.user_email = decoded.email;
+        req.body.auth0_user_id = decoded.sub;
+      }
+    });
+  }
+  next();
 }
 
-export const changeUserPassword = async function(token: string, auth0UserId: string, newPassword: string) {
+export const changeUserPassword = async function(managementToken: string, auth0UserId: string, newPassword: string) {
```
Comment (MichaelAshmead) [inline backend/lib/middleware/userInfo.ts:22]: Isn't it still a token? not the managementToken?

Comment (MichaelAshmead) [reply backend/lib/middleware/userInfo.ts:22]: Not a big issue since it's a naming change, but jw

File: backend/certificate.pem (hunk 1 above)
Comment (EduBeghe) [reply backend/certificate.pem]: Removed

**Pull request merged**